### ISSUE TITLE
Unprivileged users can take DDL locks using objects they don't own

### ISSUE THREAD

**Issue by ringerc** [2ndQuadrant/bdr#195]
Users who don't own any objects and have no rights to create them can still cause a DDL lock to be taken, disrupting all users' activity, by attempting to perform DDL on an object they don't own. The command filter tries to acquire a DDL lock before we test local permissions.

Álvaro says: "I think the grace period is reasonable protection but I wonder if we should try to avoid acquiring locks prior to checking ownership, with a callback for RangeVarGetRelidExtended which is a mechanism invented exactly for this"

A usable workaround is to `ALTER USER ... SET bdr.permit_ddl_locking = off` for such users. Coarse-grained and the user can just `SET` it away, but helps with mistakes.

Otherwise we can look at testing for ownership of the target object(s) for most DDL from the command filter, if Álvaro's approach doesn't work out.

(See RT 42364)

**Comment by PJMODOS** [comment]
I think the approach @alvherre suggested should actually be used from the command filter.

**Comment by ringerc** [comment]
Yeah, I'd like to. I have to get 1.0 out, including press releases, packages, etc, in less than a week though, and it was originally going to be today. So I think I have to bump this to 1.0.1 and fit it in concurrently with bdr96 work.

**Comment by PJMODOS** [comment]
That's fine, it's not critical issue
